test/browser.transport.test.js

```diff
@@ -31,54 +32,59 @@ describe('post', function() {
+      expect(requestFactory.request.timeout).to.equal(options.timeout);
       done(err);
     };
-    t.post(accessToken, options, payload, callback, requestFactory);
+    t.post(accessToken, options, payload, callback, requestFactory.get.bind(requestFactory));
   });
   it('should callback with the server error if 403', function(done) {
     var response = '{"err": "bad request", "result": null, "message": "fail whale"}'
-    var requestFactory = requestGenerator(response, 403);
+    var requestFactory = new requestGenerator(response, 403);
     var callback = function(err, resp) {
       expect(resp).to.not.be.ok();
       expect(err.message).to.eql('403');
+      expect(requestFactory.request.timeout).to.equal(options.timeout);
       done(resp);
     };
-    t.post(accessToken, options, payload, callback, requestFactory);
+    t.post(accessToken, options, payload, callback, requestFactory.get.bind(requestFactory));
   });
   it('should callback with the server error if 500', function(done) {
     var response = '{"err": "bad request", "result": null, "message": "500!!!"}'
-    var requestFactory = requestGenerator(response, 500);
+    var requestFactory = new requestGenerator(response, 500);
     var callback = function(err, resp) {
       expect(resp).to.not.be.ok();
       expect(err.message).to.eql('500');
+      expect(requestFactory.request.timeout).to.equal(options.timeout);
       done(resp);
     };
-    t.post(accessToken, options, payload, callback, requestFactory);
+    t.post(accessToken, options, payload, callback, requestFactory.get.bind(requestFactory));
   });
   it('should callback with a retriable error with a weird status', function(done) {
     var response = '{"err": "bad request"}'
-    var requestFactory = requestGenerator(response, 12005);
+    var requestFactory = new requestGenerator(response, 12005);
     var callback = function(err, resp) {
       expect(resp).to.not.be.ok();
       expect(err.message).to.match(/connection failure/);
       expect(err.code).to.eql('ENOTFOUND');
+      expect(requestFactory.request.timeout).to.equal(options.timeout);
       done(resp);
     };
-    t.post(accessToken, options, payload, callback, requestFactory);
+    t.post(accessToken, options, payload, callback, requestFactory.get.bind(requestFactory));
   });
   it('should callback with some error if normal sending throws', function(done) {
     var response = '{"err": "bad request"}'
-    var requestFactory = requestGenerator(response, 500, true);
+    var requestFactory = new requestGenerator(response, 500, true);
     var callback = function(err, resp) {
       expect(resp).to.not.be.ok();
       expect(err.message).to.match(/Cannot find a method to transport/);
+      expect(requestFactory.request.timeout).to.equal(options.timeout);
       done(resp);
     };
-    t.post(accessToken, options, payload, callback, requestFactory);
+    t.post(accessToken, options, payload, callback, requestFactory.get.bind(requestFactory));
```

Comment:
In each of these cases, isn't `get` already bound to `requestFactory` since it is defined on the prototype?

Reply:
No, this is not bound, sorry, I'll write a closure maybe it's "clear" than binding...